crates/re_types_builder/src/objects.rs

```diff
@@ -718,37 +692,33 @@ pub fn is_testing_fqname(fqname: &str) -> bool {
     fqname.contains("rerun.testing")
 }
 
-/// Properties specific to either structs or unions, but not both.
-#[derive(Debug, Clone)]
-pub enum ObjectSpecifics {
-    Struct,
-    Union {
-        /// The underlying type of the union.
-        ///
-        /// `None` if this is a union, some value if this is an enum.
-        utype: Option<ElementType>,
-    },
-}
-
-impl ObjectSpecifics {
-    pub fn typ(&self) -> ObjectType {
-        match self {
-            ObjectSpecifics::Struct => ObjectType::Struct,
-            ObjectSpecifics::Union { utype: None } => ObjectType::Union,
-            ObjectSpecifics::Union { utype: Some(_) } => ObjectType::Enum,
-        }
-    }
-}
-
+/// Is this a struct, enum, or union?
 #[derive(Clone, Copy, Debug, PartialEq, Eq)]
-pub enum ObjectType {
+pub enum ObjectClass {
     Struct,
 
-    /// A proper union sum type
-    Union,
-
-    /// An enumeration of alternatives, C-style.
+    /// Dumn C-style enum.
+    ///
+    /// Encoded as a sparse arrow union.
+    ///
+    /// Arrow uses a `i8` to encode the variant, forbidding negatives,
+    /// so there are 127 possible states.
+    /// We reserve `0` for a special/implicit `__null_markers` variant,
+    /// which we use to encode null values.
+    /// This means we support at most 126 possible enum variants.
+    /// Therefore the enum can be backed by a simple `u8` in Rust and C++.
```

Comment:
shouldn't we use `i8`/`int8_t` instead though then in Rust and C++?

Reply:
I don't think it matters much. Both `i8` and `u8` covers the valid range [1, 127] and are one byte big.

In Rust I currently don't set a `repr`, but I do set `: uint8_t` in C++ because 🤷